lib/ruby_indexer/lib/ruby_indexer/declaration_listener.rb

```diff
@@ -24,7 +24,7 @@ def initialize(index, dispatcher, parse_result, uri, collect_comments: false)
       @index = index
       @uri = uri
       @enhancements = T.let(Enhancement.all(self), T::Array[Enhancement])
-      @visibility_stack = T.let([Entry::Visibility::PUBLIC], T::Array[Entry::Visibility])
+      @visibility_stack = T.let([VisibilityScope.new], T::Array[VisibilityScope])
```

Comment:
Although it defaults to public, I think it would be better to be expclicit, e.g. `VisibilityScope.public` could a be factory method.